.github/compatibility-test-matrices/compat.py

```diff
@@ -0,0 +1,104 @@
+
+# JSON keys to search in.
+KEYS = ("chain-a", "chain-b")
+# Toggle if required. Indent = 4 matches jq.
+DUMP_ARGS = {"indent": 4, "sort_keys": False, "ensure_ascii": False}
+# Suggestions for recent and new versions.
+SUGGESTION = "for example, v4.3.0 or 4.3.0"
+
+def find_json_files(directory, ignores=(".",)):
+    """ Find JSON files in a directory. By default, ignore hidden directories."""
+    for root, dirs, files in os.walk(directory):
+        dirs[:] = (d for d in dirs if not d.startswith(ignores))
+        for file_ in files:
+            if file_.endswith(".json"):
+                yield os.path.join(root, file_)
+
+def has_release_version(json_file, keys, version):
+    """ Check if the json file has the version in question. """
+    return any(version in row for row in (json_file[key] for key in keys))
+
+def update_version(json_file, keys, args):
+    """ Update the versions as required in the json file. """
+    recent, new, op = args.recent, args.new, args.type
+    for row in (json_file[key] for key in keys):
+        if recent not in row:
+            continue        
+        if op == "add":
+            row.append(new)
+            row.sort(reverse=True)
+        else:
+            index = row.index(recent)
+            if op == "replace":
+                row[index] = new
+            elif op == "remove":
+                del row[index]
+
+def version_input(prompt, version):
+    """ Input version if not supplied, make it start with a 'v' if it doesn't."""
+    if version is None:
+        version = input(prompt)
+    return version if version.startswith(("v", "V")) else f"v{version}"
+
+def require_version(args):
+    """ Allow non-required version in argparse but request it if not provided."""
+    args.recent = version_input(f"Recent version ({SUGGESTION}): ", args.recent)
+    if args.type == "remove":
+        return
+    args.new = version_input(f"New version ({SUGGESTION}): ", args.new)
+
+def parse_args():
+    """ Parse command line arguments."""
+    parser = argparse.ArgumentParser(description="Update JSON files.")
+    parser.add_argument(
+        "--type",
+         choices=["add", "replace", "remove"], 
+         default="add",
+         help="Type of version update: add a version, replace one or remove one."
+    )
+    parser.add_argument(
+        "--directory", 
+        default=".", 
+        help="Directory path where JSON files are located"
+    )
+    parser.add_argument(
+        "--recent_version",
+        dest="recent",
+        help=f"Recent version to search in JSON files ({SUGGESTION})"
+    )
+    parser.add_argument(
+        "--new_version", 
+        dest="new",
+        help=f"New version to add in JSON files ({SUGGESTION})"
+    )
+    parser.add_argument(
+        "--verbose",
+        "-v",
+        action="store_true",
+        help="Allow for verbose output",
+        default=False,
+    )
+
+    args = parser.parse_args()
+    require_version(args)
+    return args
+
+def main(args):
```

Comment:
one thing I noticed was that when we run the script giving an version that doesn't exists, it executes successfully but is just silent. I think it would be no harm either logging that the version wasn't found, or raise a ValueError with the input not having been found.

Reply:
yeah, can see the value in that